### PR TITLE
fix(module:tag): Issue solved & align with react antD & tests

### LINKED ISSUE #1626: Question: is that possible to prevent OnClick when OnClosing in Tag component?
I have a Tag to support to search link by click the tag and remove the tag by closing it.
However, when I try to close the tag, it fire OnClick and OnClosing at the same time.
I wonder that it is possible to prevent OnClick when OnClosing ?

```csharp
<Tag Closable="true"
     OnClosing="e => OnRemovingTag(bk.Bk, tag)"
     OnClick="e => OnClickTag(bk.Bk, tag)"
     Color="@WebHelper.GetTagColor(tag)">
    @tag
</Tag>
```

EYh8qwg5aR

### PR DESCRIPTION
**Pull request by anddrzejb**
### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
Fixes #1626

### 💡 Background and solution
1. Fixing #1626 required just to change the order of calls.
2. Added new parameter `Animate`. This is not the most elegant solution. Unfortunately, I am not very skilled when it comes to animations. I do hope some day someone will make it better, but at least at this stage it seems to be working, so that beats the lack of functionality. I used here css animations only with css isolation. Please inspect if this is accepted way of doing things. There seems though that docs site does not see the css - that is why in `Animate.razor` demo I actually embedded the animation styles. During my tests, when using in project, css in `Tag.razor.css` was picked up properly (both server-side and wasm). 
3. Added custom color handling.
4. Added new color for `success`, `processing`, `warning`, `error` & `default`
5. Added xml comments
6. Added unit tests.

### 📝 Changelog
| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Added `Animate` parameter. Added new status tags : `success`, `processing`, `warning`, `error` & `default`. Added custom color handling.         |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by ElderJames** [comment]
Thanks for the work!
The animations css should put in the `components/tag/style/patch.less`, and import it into `components/tag/style/entry.less`. Because css isolation cannot be merged into the `ant-design-blazor.css` file.

**Comment by anddrzejb** [comment]
@ElderJames Most of things here were low hanging fruits (maybe except animation, but probably only to me as I do not have much experience in that area), but I figured they still needed to be done. 
Thanks for the directions on how to do custom styling! Helped a lot.

**Comment by ElderJames** [comment]
@anddrzejb Sorry, I found that the animation feature in react impl is not built-in, and just a demo with third-party library. So let's not make it a built-in feature, because users will probably want to use other animation effects. We can just add the demo with animation css to show how to use animation.

**Comment by anddrzejb** [comment]
As you wish, however current implementation is not only css but also the delays hardcoded in `Tag` (have a look at `OnAfterRenderAsync` especially line 128 and `CloseTag` especially line 191). I guess we could parametrize them, then consumers could easily use them.

**Comment by ElderJames** [comment]
I think we don't really need to delay the close, but just leave it open and the user can use the `OnClose` event to hide the tag themselves.

**Comment by anddrzejb** [comment]
Ok, I will try to remove the animation related code and prep a demo that will show how to extend the `Tag` component to actually have one.

File: components/tag/Tag.razor.cs [hunk 1]
```diff
@@ -20,28 +51,34 @@ public string Color
...
-
-        [Parameter]
-        public bool Checkable { get; set; }
-
-        [Parameter]
-        public bool Visible { get; set; } = true;
-
-        [Parameter]
-        public bool Checked { get; set; }
+        public string ExtraCls { get; set; } = "";
```
[the remark this answers is not in the document]
Comment (anddrzejb) [reply components/tag/Tag.razor.cs]: I have forgotten about that property. Thanks for reminding me. Commit fe5e3cd resolves this.

File: components/tag/Tag.razor [hunk 2]
```diff
@@ -3,15 +3,15 @@
 
 @if (!_closed)
 {
-    <span class="@ClassMapper.Class" style="@Style" id="@Id" @ref="Ref" @onclick="ClickTag">
+    <span class="@ClassMapper.Class @Class" style="@_style" id="@Id" @ref="Ref" @onclick="ClickTag">
```
Comment (ElderJames) [inline components/tag/Tag.razor:6]:
```suggestion
    <span class="@ClassMapper.Class" style="@_style" id="@Id" @ref="Ref" @onclick="ClickTag">
```

**Review by ElderJames — APPROVED** [review]

**Pull request merged**